elastic/kibana
# [RAC] turn off observability alerts as data writing in a more granular way

Fixes https://github.com/elastic/kibana/issues/119217

`xpack.ruleRegistry.write.disabledRegistrationContexts` flag was introduced to disable writing to observability alerts-as-data indices in a more granular way.

The registration contexts we use are:

- observability.logs
- observability.metrics
- observability.apm
- observability.uptime

### How to test
- In kibana.dev.yml create a new config `xpack.ruleRegistry.write.disabledRegistrationContexts : ['observability.logs']` (you could try with other values from the above list)
- Delete alerts indices by restarting ES and kibana in a local setup. If you use a CCS setup make sure you stop Kibana and delete alerts indices, index templates and component templates. Here's an example of how you could reset your cluster
```
http DELETE 'https://YOUR_ENDPOINT/_index_template/.alerts*' && 
http DELETE 'https://YOUR_ENDPOINT/_component_template/.alerts*' && 
http DELETE 'https://YOUR_ENDPOINT/.kibana*,.internal*,.tasks*'
```
- Restart kibana 
- Create a new rule with sensitive thresholds for log threshold (if you specified another registration context in kibana.dev.yml create a rule of that type accordingly)
- Verify that writing to the specified registration context(s) is disabled. You could verify this under Kibana > Stack Management > Index Management. 
  - Under `Indices` tab make sure to enable the `Include hidden indices` toggle, search for `.internal.alerts` and verify that nothing appears on the list. 
  - Under `Index templates` tab, enable `View System templates`, search for `alerts` and verify that nothing appears on the list. 
  - Verify that the specified disabled registration contexts don't appear under `Component templates`
- You could also verify that disabling writing to specified registration contexts works, by creating rules for the specified registration contexts and making sure that no alerts appear on the Alerts table
